cmd/build/v2/build.go

```diff
@@ -242,7 +242,18 @@ func (bc *OktetoBuilder) addVolumeMounts(ctx context.Context, manifest *model.Ma
 	if err != nil {
 		return "", err
 	}
-	buildOptions := build.OptsFromBuildInfo(manifest.Name, svcName, svcBuild, &types.BuildOptions{})
+	buildOptions := build.OptsFromBuildInfo(manifest.Name, svcName, svcBuild, options)
+
+	if build.ShouldOptimizeBuild(buildOptions) {
```

Comment:
Not sure we should also apply the optimization at the volume-mounts image when pipeline? this looks for a tag not the hash so i'm afraid the tag will be built but for other base image. maybe i'm having missunderstandings about the volumes here 🤷🏼‍♀️.

cc: @pchico83 , @jLopezbarb

Reply:
I don't think that the built image comes from another image, with checking that it is built I think it would be enough, because if the two images are built one after the other if there is an image and the other will also come from the last built image of the first image (I think I have explained myself terribly but I hope it is understood)😅

Reply:
understood :D